### DISCUSSION TITLE
How do I inspect the full debug/launch command line?

### DISCUSSION THREAD

**Discussion by guw** [microsoft/vscode-java-debug#1438] [Q&A] [answered]
I am setting `javaExec` on my launch configuration and I suspect something is not applied correctly. How can I see the final command line that VS Code used for launching the app in debug mode (include the debug ports, etc.)?

**Comment by testforstephen** [comment, marked as answer]
By default, Java debugger uses the integrated terminal to launch your app, and the terminal will echo the full launch command line there. The launching console is controlled by the setting `"java.debug.settings.console": "integratedTerminal"`.
